lib/format-bar.js

```diff
@@ -1,11 +1,26 @@
+let platformSpace = 0;
+if (process.platform === 'win32'){
+  platformSpace = 1;
+}
+
 // format bar
 module.exports = function formatBar(progress, options){
     // calculate barsize
-    const completeSize = Math.round(progress*options.barsize);
-    const incompleteSize = options.barsize-completeSize;
-
-   // generate bar string by stripping the pre-rendered strings
-   return   options.barCompleteString.substr(0, completeSize) +
-            options.barGlue +
-            options.barIncompleteString.substr(0, incompleteSize);
-}
\ No newline at end of file
+    let columns = options.barsize;
+    if (options.autoExpand) {
+      columns = (options.stream && options.stream.columns) ? options.stream.columns : columns;
+      columns = columns - options.restWidth - platformSpace;
+
+      // dynamic width handling, on window resize and change of other context parameters
+      options.barCompleteString = (new Array(columns + 1 ).join(options.barCompleteChar));
```

Comment:
The strings should be pre-computed for performance reasons

Reply:
Could only pre compute it with a huge length, cause we don't know big the screen is. I think performance is not really a problem of the progressbar.